### PR TITLE
Hooks

### PR DESCRIPTION
**Pull request by hagenburger**

### CODE REVIEW

File: test/document.test.js [hunk 1]
```diff
@@ -157,6 +157,39 @@ describe("Options", () => {
...
+      test
+      ~~~
+    `
+    let doc = new Document(input)
+    doc.hooks.add("example:beforeRender", "add h1", function(example) {
+      example.dom.innerHTML = "<h1>"
+    })
+
+    let html = doc.render()
+    expect(html).to.have.selector("h1")
```
Comment (benevbright) [inline test/document.test.js]: I printed log of `doc.render()` and it gives an empty node with only `h1` tag.

>   Hooks
> `[doc.render()] <h1></h1>`
>     ✓ should run hooks
>     ✓ should run hooks defined in Config

Comment (hagenburger) [reply test/document.test.js]: Right, that test example should be improved.

Comment (benevbright) [reply test/document.test.js]: Ah, I misunderstood. Yes, it was working correctly though.

**Comment by benevbright** [comment]
Great work! 👍

**Pull request merged**